### ISSUE TITLE
Transduce does more work than necessary

### ISSUE THREAD

**Issue by cristianocd** [ramda/ramda#1792]
Changing the function in the example http://ramdajs.com/docs/#transduce to print each time it's been called:
```
var addOne = x => {
  console.log('add one to', x);
  return x + 1;
}

var numbers = [1, 2, 3, 4];
var transducer = R.compose(R.map(addOne), R.take(2));

R.transduce(transducer, R.flip(R.append), [], numbers); //=> [2, 3]
```
Console prints:
```
add one to
1
add one to
2
add one to
3
```
I see #921 had the same behaviour and was supposed to be closed by #923 , but the extra step persists.

**Comment by kedashoe** [comment]
Hi @cristianocd ,

You're transducer is mapping before it is taking. See eg #1790. You can either change the order of your arguments to `R.compose` or use `R.pipe` instead of `R.compose` to get the behavior you are looking for

http://goo.gl/tvU9w9

**Comment by cristianocd** [comment]
Hi @kedashoe ,
Please, see https://gist.github.com/cristianocd/914ffa9356247d7751508ce8e146f05b for a more complete example.

If I were to take before anything else, the result would be an empty array.

Thanks for the fast response

**Comment by kedashoe** [comment]
Transducers do not actually know anything about lazyness. See for example this answer http://stackoverflow.com/a/25656422/342588 (sorry I couldn't find any more on it at the moment). There has been much talk about lazyness with ramda, but as far as I know there is nothing actively being done on that front at the moment.

**Comment by cristianocd** [comment]
That's not the point. 
It's the same as #921  

@CrossEye also has observed this problem , in this comment https://github.com/ramda/ramda/issues/1304#issuecomment-123932150 , he states:
> For some reason I really should investigate, this runs through until the fourth match on the filter and then stops, having included only the three. Something seems very wrong with that, but it still has the overall laziness you're looking for.

**Comment by kedashoe** [comment]
Ah, my apologies, I think you are correct :) It looks like the fix for #921 was reverted in #1245 . Would you like to submit a PR?

**Comment by cristianocd** [comment]
Sure, no problem :)
I will create the missing test case.
